tests/integration/cassandra_auth_test.go

```diff
@@ -0,0 +1,149 @@
+//go:build integration
+// +build integration
+
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"github.com/newrelic/nri-cassandra/tests/integration/jsonschema"
+	"github.com/newrelic/nri-cassandra/tests/integration/testutils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"github.com/testcontainers/testcontainers-go"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+const (
+	SSLConnectionTimeout = "10000"
+)
+
+type CassandraSSLTestSuite struct {
+	suite.Suite
+	compose *testcontainers.LocalDockerCompose
+}
+
+func TestCassandraSSLTestSuite(t *testing.T) {
+	suite.Run(t, new(CassandraSSLTestSuite))
+}
+
+func (s *CassandraSSLTestSuite) SetupSuite() {
+	s.compose = testutils.ConfigureSSLCassandraDockerCompose()
+
+	err := testutils.RunDockerCompose(s.compose)
+	require.NoError(s.T(), err)
+
+	// Could not rely on testcontainers wait strategies here, as the server might be up but not reporting all mbeans.
+	s.T().Log("Wait for cassandra to initialize...")
+	time.Sleep(30 * time.Second)
+}
+
+func (s *CassandraSSLTestSuite) TearDownSuite() {
+	s.compose.Down()
+}
+
+func (s *CassandraSSLTestSuite) TestCassandraIntegration_SSL() {
+	t := s.T()
+
+	testName := testutils.GetTestName(t)
+
+	ctx, cancelFn := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancelFn()
+
+	env := map[string]string{
+		"METRICS": "true",
+		"TIMEOUT": SSLConnectionTimeout,
+
+		"HOSTNAME": testutils.Hostname,
+		"USERNAME": testutils.JMXUsername,
+		"PASSWORD": testutils.JMXPassword,
+
+		"TRUST_STORE":          "/certs/cassandra.truststore",
```

Comment:
would make sense to have this as well in testutils as constants?

Same comments for below strings

Reply:
good point, I'll move them